internal/libyaml/api_test.go

```diff
@@ -0,0 +1,527 @@
+	switch v := arg.(type) {
+	case bool:
+		return v
+	case string:
+		return v == "true"
+	default:
+		panic(fmt.Sprintf("cannot parse arg as bool: %v", arg))
+	}
+}
+
+// parseStringArg parses a string argument
+func parseStringArg(arg interface{}, useBytes bool) string {
+	if str, ok := arg.(string); ok {
+		return str
+	}
+	panic(fmt.Sprintf("cannot parse arg as string: %v", arg))
+}
+
+// looksLikeConstant checks if a string looks like a constant name
+func looksLikeConstant(s string) bool {
+	if len(s) == 0 {
+		return false
+	}
+	// Check if it's all uppercase letters, digits, and underscores
+	for _, c := range s {
+		if !((c >= 'A' && c <= 'Z') || (c >= '0' && c <= '9') || c == '_') {
+			return false
+		}
+	}
+	return true
+}
+
+// parseConstant parses a constant name to its integer value
+func parseConstant(name string) int {
+	// Handle boolean values
+	if name == "true" {
+		return 1
+	}
+	if name == "false" {
+		return 0
+	}
+
+	// Try parsing as int first
+	if val, err := strconv.Atoi(name); err == nil {
+		return val
+	}
+
+	// Use IntOrStr to parse other constants
+	ios := IntOrStr{}
+	err := ios.FromValue(name)
+	if err != nil {
+		panic(fmt.Sprintf("failed to parse constant %q: %v", name, err))
+	}
+	return ios.Value
+}
+
+// runFieldChecks runs field checks on an object
+func runFieldChecks(t *testing.T, obj interface{}, checks []FieldCheck) {
+	t.Helper()
+
+	v := reflect.ValueOf(obj)
+	if v.Kind() == reflect.Ptr {
+		v = v.Elem()
+	}
+
+	for _, check := range checks {
+		// Handle nil checks
+		if len(check.Nil) >= 2 {
+			fieldName, _ := check.Nil[0].(string)
+			wantNil, _ := check.Nil[1].(bool)
+			field := getField(t, v, fieldName)
+			if field.IsValid() {
+				isNil := field.IsNil()
+				if wantNil != isNil {
+					if wantNil {
+						t.Errorf("%s should be nil", fieldName)
+					} else {
+						t.Errorf("%s should not be nil", fieldName)
+					}
+				}
+			}
+		}
+
+		// Handle cap checks
+		if len(check.Cap) >= 2 {
+			fieldName, _ := check.Cap[0].(string)
+			wantCap, _ := check.Cap[1].(int)
+			field := getField(t, v, fieldName)
+			if field.IsValid() && wantCap > 0 {
+				if field.Cap() != wantCap {
+					t.Errorf("%s cap = %d, want %d", fieldName, field.Cap(), wantCap)
+				}
+			}
+		}
+
+		// Handle len checks
+		if len(check.Len) >= 2 {
+			fieldName, _ := check.Len[0].(string)
+			wantLen, _ := check.Len[1].(int)
+			field := getField(t, v, fieldName)
+			if field.IsValid() && wantLen > 0 {
+				if field.Len() != wantLen {
+					t.Errorf("%s len = %d, want %d", fieldName, field.Len(), wantLen)
+				}
+			}
+		}
+
+		// Handle len-gt checks
+		if len(check.LenGt) >= 2 {
+			fieldName, _ := check.LenGt[0].(string)
+			minLen, _ := check.LenGt[1].(int)
+			field := getField(t, v, fieldName)
+			if field.IsValid() && minLen > 0 {
+				if field.Len() <= minLen {
+					t.Errorf("%s len = %d, want > %d", fieldName, field.Len(), minLen)
+				}
+			}
+		}
+
+		// Handle eq checks
+		if len(check.Eq) >= 2 {
```

Comment:
Is there a valid case when having more than 2 arguments shouldn't be considered as an error?

The test suite shouldbe human error prone